tests/test_bco_trainer.py

```diff
@@ -20,76 +20,83 @@
+from transformers import AutoModel, AutoModelForCausalLM, AutoTokenizer
 from transformers.testing_utils import require_peft
+from transformers.utils import is_peft_available
 
 from trl import BCOConfig, BCOTrainer
 from trl.trainer.bco_trainer import _process_tokens, _tokenize
 
 from .testing_utils import require_no_wandb, require_sklearn
 
 
+if is_peft_available():
+    from peft import LoraConfig
+
+
 class BCOTrainerTester(unittest.TestCase):
-    def setUp(self):
-        self.model_id = "trl-internal-testing/tiny-Qwen2ForCausalLM-2.5"
-        self.model = AutoModelForCausalLM.from_pretrained(self.model_id)
-        self.ref_model = AutoModelForCausalLM.from_pretrained(self.model_id)
-        self.tokenizer = AutoTokenizer.from_pretrained(self.model_id)
-        self.tokenizer.pad_token = self.tokenizer.eos_token
-
-        # get t5 as seq2seq example:
-        model_id = "trl-internal-testing/tiny-T5ForConditionalGeneration"
-        self.t5_model = AutoModelForSeq2SeqLM.from_pretrained(model_id)
-        self.t5_ref_model = AutoModelForSeq2SeqLM.from_pretrained(model_id)
-        self.t5_tokenizer = AutoTokenizer.from_pretrained(model_id)
-
-        # get embedding model
-        model_id = "trl-internal-testing/tiny-BartModel"
-        self.embedding_model = AutoModel.from_pretrained(model_id)
-        self.embedding_tokenizer = AutoTokenizer.from_pretrained(model_id)
-
-    @parameterized.expand(
-        [
-            ("qwen", True, True, "standard_unpaired_preference"),
-            ("qwen", True, False, "standard_unpaired_preference"),
-            ("qwen", False, True, "standard_unpaired_preference"),
-            ("qwen", False, False, "standard_unpaired_preference"),
-            ("qwen", True, True, "conversational_unpaired_preference"),
-        ]
-    )
+    @parameterized.expand([("standard_unpaired_preference"), ("conversational_unpaired_preference")])
     @require_sklearn
-    def test_bco_trainer(self, name, pre_compute, eval_dataset, config_name):
+    def test_train(self, config_name):
+        model_id = "trl-internal-testing/tiny-Qwen2ForCausalLM-2.5"
+        model = AutoModelForCausalLM.from_pretrained(model_id)
+        ref_model = AutoModelForCausalLM.from_pretrained(model_id)
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+        dataset = load_dataset("trl-internal-testing/zen", config_name, split="train")
+
         with tempfile.TemporaryDirectory() as tmp_dir:
             training_args = BCOConfig(
                 output_dir=tmp_dir,
-                per_device_train_batch_size=2,
-                max_steps=3,
-                gradient_accumulation_steps=1,
-                learning_rate=9e-1,
-                eval_strategy="steps" if eval_dataset else "no",
-                beta=0.1,
-                precompute_ref_log_probs=pre_compute,
+                remove_unused_columns=False,  # warning raised if not set to False
```

Comment:
Not for this PR, but should we then set this as the default value in `BCOConfig`?

Reply:
Or even better, make BCO support this arg. Like we did for DPO in https://github.com/huggingface/trl/pull/2233